File: arena-junit/src/test/java/arena/junit/dep/temporal/TemporalDependencyBuilderSerializationTest.java
package arena.junit.dep.temporal;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

import com.fasterxml.jackson.databind.node.ObjectNode;

import org.junit.jupiter.api.Test;

final class TemporalDependencyBuilderSerializationTest {

  @Test
  void build_minimalName_serializesTypeAndIdentifier() {
    ObjectNode config = new TemporalDependencyBuilder("temporal").build().forFfi();
    assertEquals("temporal", config.path("type").asText());
    assertTrue(config.path("identifier").asText().startsWith("arena-temporal-temporal-"));
    assertFalse(config.has("image"));
    assertFalse(config.has("port"));
  }

  @Test
  void withOverrides_allFields_serializesConfiguredFields() {
    ObjectNode config =
        new TemporalDependencyBuilder("temporal")
            .withImage("1.24.2")
            .withImageName("temporalio/auto-setup")
            .withPort(17233)
            .withUiPort(18233)
            .withContainerName("temporal-box")
            .build()
            .forFfi();
    assertEquals("1.24.2", config.path("image").asText());
    assertEquals("temporalio/auto-setup", config.path("image_name").asText());
    assertEquals(17233, config.path("port").asInt());
    assertEquals(18233, config.path("ui_port").asInt());
    assertEquals("temporal-box", config.path("container_name").asText());
  }
}

File: arena-xunit/tests/Unit/TemporalDependencyBuilderSerializationTest.cs
using ArenaXunit.Dep;
using Newtonsoft.Json.Linq;
using Xunit;

namespace ArenaXunit.UnitTest;

public class TemporalDependencyBuilderSerializationTest
{
    [Fact]
    public void Build_DefaultPort_SerializesCorrectJson()
    {
        var dep = new TemporalDependencyBuilder("test").Build();
        var json = dep.ForFfi();
        var obj = JObject.Parse(json);
        Assert.Equal("temporal", obj["type"]);
        Assert.Equal(7233, obj["port"]);
        Assert.NotNull(obj["identifier"]);
    }

    [Fact]
    public void Build_CustomPort_SerializesCorrectJson()
    {
        var dep = new TemporalDependencyBuilder("test").WithPort(7333).Build();
        var json = dep.ForFfi();
        var obj = JObject.Parse(json);
        Assert.Equal(7333, obj["port"]);
    }

    [Fact]
    public void Build_DefaultUiPort_SerializesCorrectJson()
    {
        var dep = new TemporalDependencyBuilder("test").Build();
        var json = dep.ForFfi();
        var obj = JObject.Parse(json);
        Assert.Equal(8233, obj["ui_port"]);
    }

    [Fact]
    public void Build_CustomUiPort_SerializesCorrectJson()
    {
        var dep = new TemporalDependencyBuilder("test").WithUiPort(9333).Build();
        var json = dep.ForFfi();
        var obj = JObject.Parse(json);
        Assert.Equal(9333, obj["ui_port"]);
    }

    [Fact]
    public void Build_WithImageAndContainerName_SerializesCorrectJson()
    {
        var dep = new TemporalDependencyBuilder("test")
            .WithImage("1.8.0")
            .WithImageName("temporalio/temporal")
            .WithContainerName("my-temporal")
            .Build();
        var json = dep.ForFfi();
        var obj = JObject.Parse(json);
        Assert.Equal("1.8.0", obj["image"]);
        Assert.Equal("temporalio/temporal", obj["image_name"]);
        Assert.Equal("my-temporal", obj["container_name"]);
    }

    [Fact]
    public void Build_Identifier_MatchesPattern()
    {
        var dep = new TemporalDependencyBuilder("test").Build();
        Assert.StartsWith("arena-temporal-", dep.Identifier);
    }
}
